src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -185,8 +185,15 @@ private void MarkPropertiesAsSkipped(string currentModelKey, ModelMetadata metad
                     Visited = validationContext.Visited
                 };
 
-                var propertyBindingName = propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName;
-                var childKey = ModelBindingHelper.CreatePropertyModelName(currentModelKey, propertyBindingName);
+                // For a body bound model, we do not include the entire chain upto the body.
+                // Instead we start with an empty prefix.
+                var childKey = string.Empty;
```

Comment:
This is a temporary fix: The best way of solving these is to reintroduce the validation node concept to pass data between the modelbinding and validation systems.